myokit/_model_api.py

```diff
@@ -1659,6 +1659,276 @@ def has_warnings(self):
+                    except KeyError:
+                        raise myokit.WellMappedError(
+                            self_var + ' does not exist in this model'
+                        )
+                else:
+                    raise TypeError(
+                        'Variables in the var_map need to be specified as ' +
+                        'objects or as fully qualified names'
+                    )
+
+                # check external variables in var_map are variables and exists
+                # in external model
+                if isinstance(ext_var, myokit.Variable):
+                    if not ext_var.has_ancestor(external_component.model()):
+                        raise myokit.WellMappedError(
+                            ext_var.name() +
+                            ' does not exist in the external component\'s' +
+                            'model'
+                        )
+                    ext_var = ext_model_copy.var(ext_var.qname())
+                elif isinstance(ext_var, basestring):
+                    try:
+                        ext_var = ext_model_copy.var(ext_var)
+                    except KeyError:
+                        raise myokit.WellMappedError(
+                            ext_var +
+                            ' does not exist in the external component\'s' +
+                            'model'
+                        )
+                else:
+                    raise TypeError(
+                        'Variables in the var_map need to be specified as ' +
+                        'objects or as fully qualified names'
+                    )
+
+                # check for duplicates in var_map
+                if self_var in copied_var_list:
+                    raise myokit.WellMappedError(
+                        'Multiple variables map onto ' + self_var.name()
+                    )
+
+                # add to full_var_map if a relevant variable
+                if ext_var in relevant_vars:
+                    full_var_map[ext_var] = self_var
+                copied_var_list.append(self_var)
+
+        for ext_var in relevant_vars:
+            if ext_var not in copied_var_list:
+                ext_bind = ext_var.binding()
+                ext_label = ext_var.label()
+                self_bind_var = self.binding(ext_bind)
+                self_label_var = self.label(ext_label)
+
+                if self_bind_var is not None:
+                    # add self_bind_var:ext_var to full_var_map
+                    full_var_map[ext_var] = self_bind_var
+                elif self_label_var is not None:
+                    # add self_label_var:ext_var to full_var_map
+                    full_var_map[ext_var] = self_label_var
+                elif allow_name_mapping:
+                    if self.has_variable(ext_var.qname()):
+                        # add to full map
+                        full_var_map[ext_var] = self.get(ext_var.qname())
+            if (
```

Comment:
What about the name mapping?

Reply:
The name mapping is there. lines 1825 to 1828
```
    elif allow_name_mapping:
        if self.has_variable(ext_var.qname()):
            # add to full map
            full_var_map[ext_var] = self.get(ext_var.qname())
```